selenium.js

```diff
@@ -517,40 +523,41 @@ const runAll = async (limitBrowsers, oses, nonConcurrent, reverse) => {
 };
 
 /* istanbul ignore if */
-if (require.main === module) {
-  const {argv} = require('yargs').command(
-      '$0 [browser..]',
-      'Run Selenium on several browser versions',
-      (yargs) => {
-        yargs
-            .positional('browser', {
-              describe: 'Limit the browser(s) to test',
-              type: 'string',
-              choices: ['chrome', 'edge', 'firefox', 'ie', 'safari']
-            })
-            .option('os', {
-              describe: 'Specify OS to test',
-              type: 'array',
-              choices: ['Windows', 'macOS'],
-              default: ['Windows', 'macOS']
-            })
-            .option('non-concurrent', {
-              describe: 'Run browsers sequentially (one at a time)',
-              alias: 's',
-              type: 'boolean',
-              nargs: 0
-            })
-            .option('reverse', {
-              describe: 'Run browser versions oldest-to-newest',
-              alias: 'r',
-              type: 'boolean',
-              nargs: 0
-            });
-      }
+if (esMain(import.meta)) {
+  const {argv} = yargs(hideBin(process.argv)).command(
+    '$0 [browser..]',
+    'Run Selenium on several browser versions',
+    (yargs) => {
+      yargs
+        .positional('browser', {
+          describe: 'Limit the browser(s) to test',
+          type: 'string',
+          choices: ['chrome', 'edge', 'firefox', 'ie', 'safari']
+        })
+        .option('os', {
+          describe: 'Specify OS to test',
+          type: 'array',
+          choices: ['Windows', 'macOS'],
+          default: ['Windows', 'macOS']
+        })
+        .option('non-concurrent', {
+          describe: 'Run browsers sequentially (one at a time)',
+          alias: 's',
+          type: 'boolean',
+          nargs: 0
+        })
+        .option('reverse', {
+          describe: 'Run browser versions oldest-to-newest',
+          alias: 'r',
+          type: 'boolean',
+          nargs: 0
+        });
+    }
   );
 
   if (
-    runAll(argv.browser, argv.os, argv.nonConcurrent, argv.reverse) === false
+    (await runAll(argv.browser, argv.os, argv.nonConcurrent, argv.reverse)) ===
```

Comment:
Since a rejected promise is now automatically turned into a non-zero exit code, can we instead throw exceptions in `runAll()` for the cases where we return `false` now and just `await runAll(argv.browser, argv.os, argv.nonConcurrent, argv.reverse)` here instead?